File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/javascript/recursion/zigzag.js
/**
 * https://www.codestepbystep.com/problem/view/javascript/recursion/zigzag
 *
 * @author Alex Golub
 * @since 09-May-21, 3:05 PM
 */
function zigzag(n) {
  if (n < 1) {
    throw 'argument must be greater than or equal to 1';
  }
  console.log(helper(n, ''));
}

function helper(n, accum) {
  if (n === 1) {
    return '*';
  } else if (n === 2) {
    return '**';
  } else {
    return '<' + helper(n - 2, accum) + '>';
  }
}

zigzag(1);
zigzag(2);
zigzag(3);
zigzag(4);

File: Back to Basics Approach 3rd Edition/CodeStepByStep_java/recursion/zigzag.java
package CodeStepByStep.java.recursion;

/**
 * https://www.codestepbystep.com/problem/view/java/recursion/zigzag
 *
 * @author Alex Golub
 * @since 09-May-21, 3:01 PM
 */
class zigzag {
  public static void zigzag(int n) {
    if (n < 1) {
      throw new IllegalArgumentException();
    } else if (n == 1) {
      System.out.print("*");
    } else if (n == 2) {
      System.out.print("**");
    } else {
      System.out.print("<");
      zigzag(n - 2);
      System.out.print(">");
    }
  }
}
